tests/test_compose_ownership.py

```
@@ -55,11 +57,54 @@ def test_notebook_service_has_an_explicit_command_after_entrypoint_override() ->
+def test_tini_download_is_checksum_verified() -> None:
+    dockerfile = DOCKERFILE.read_text(encoding="utf-8")
+
+    assert "TINI_VERSION=v0.19.0" in dockerfile
+    assert "93dcc18adc78c65a028a84799ecf8ad40c936fdfc5f2a57b1acda5a8117fa82c" in dockerfile
+    assert 'echo "${TINI_SHA256}  /tini" | sha256sum -c - && chmod +x /tini' in dockerfile
+
+
+def test_compose_rendering_preserves_service_bindings() -> None:
+    if shutil.which("docker") is None:
+        pytest.skip("Docker Compose is not installed")
+    rendered = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)